src/test/java/org/apache/maven/buildcache/its/RemoteCacheDavTest.java

```diff
@@ -0,0 +1,493 @@
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.FileTime;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.maven.buildcache.its.junit.BeforeEach;
+import org.apache.maven.buildcache.its.junit.Inject;
+import org.apache.maven.buildcache.its.junit.IntegrationTest;
+import org.apache.maven.buildcache.its.junit.IntegrationTestExtension;
+import org.apache.maven.it.VerificationException;
+import org.apache.maven.it.Verifier;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+@IntegrationTest( "src/test/projects/remote-cache-dav" )
+@Testcontainers
+@EnabledOnOs( OS.LINUX ) // github actions do not support docker on windows and osx
+public class RemoteCacheDavTest
+{
+
+    public static final String DAV_DOCKER_IMAGE = "xama/nginx-webdav@sha256:84171a7e67d7e98eeaa67de58e3ce141ec1d0ee9c37004e7096698c8379fd9cf";
+    private static final String DAV_USERNAME = "admin";
+    private static final String DAV_PASSWORD = "admin";
+    private static final String REPO_ID = "build-cache";
+    private static final String HTTP_TRANSPORT_PRIORITY = "aether.priority.org.eclipse.aether.transport.http.HttpTransporterFactory";
+    private static final String WAGON_TRANSPORT_PRIORITY = "aether.priority.org.eclipse.aether.transport.wagon.WagonTransporterFactory";
+    private static final String MAVEN_BUILD_CACHE_REMOTE_SAVE_ENABLED = "maven.build.cache.remote.save.enabled";
+
+    @Container
+    GenericContainer<?> dav;
+
+    @Inject
+    Verifier verifier;
+
+    Path basedir;
+    Path remoteCache;
+    Path localCache;
+    Path settings;
+    Path logDir;
+
+    @BeforeEach
+    void setup() throws IOException
+    {
+        assumeTrue( DockerClientFactory.instance().isDockerAvailable(), "docker is not available" );
```

Comment:
just use `@Testcontainers(disabledWithoutDocker = true)` at class level and you do not need that